cmd/fleet/main.go

```diff
@@ -65,47 +65,54 @@ func runBulkActions(ctx context.Context, sv saved.CRUD) *BulkActions {
 	return ba
 }
 
-func runPolicyMon(ctx context.Context, sv saved.CRUD) *PolicyMon {
+func runPolicyMon(ctx context.Context, sv saved.CRUD) (*PolicyMon, error) {
 	pm, err := NewPolicyMon(kPolicyThrottle)
-	checkErr(err)
-
+	if err != nil {
+		return nil, err
+	}
 	go func() {
-		err := pm.Monitor(ctx, sv)
-		checkErr(err)
+		for {
+			err := pm.Monitor(ctx, sv)
+			if err == context.Canceled {
+				break
+			}
+			log.Error().Err(err).Msg("Restart policy monitor on error")
+		}
 	}()
-
-	return pm
+	return pm, err
 }
 
-func runActionMon(ctx context.Context, bulker bulk.Bulk) *action.Monitor {
+func runActionMon(ctx context.Context, bulker bulk.Bulk) (*action.Monitor, error) {
 	am, err := action.NewMonitor(bulker)
 	if err != nil {
-		checkErr(err)
+		return nil, err
 	}
-
 	go func() {
-		err := am.Run(ctx)
-		checkErr(err)
+		for {
+			err := am.Run(ctx)
+			if err == context.Canceled {
+				break
+			}
+			log.Error().Err(err).Msg("Restart action monitor on error")
+		}
 	}()
-
-	return am
+	return am, nil
 }
 
 func runActionDispatcher(ctx context.Context, am *action.Monitor) *action.Dispatcher {
 	ad := action.NewDispatcher(am)
-
 	go func() {
-		err := ad.Run(ctx)
-		checkErr(err)
+		for {
+			err := ad.Run(ctx)
+			if err == context.Canceled {
+				break
+			}
+			log.Error().Err(err).Msg("Restart action dispatcher on error")
+		}
```

Comment:
This could cause tight loops on errors on Run of server different subcomponents, difficult to manage state when some of the components started and some are in the tight running loops trying to start.

maybe should consider the fleet server as one restartable unit and if any of its components fail then exit the server and have some kind of restart with backoff logic for the server restart.
I can address in the next PR, did something similar for Endgame event streaming before.

Reply:
i think if we handle the restart as a single unit, that would be good